JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java

```diff
@@ -257,58 +273,144 @@ private Behavior<Event> getGoldenIdsHandler(final GetGoldenIdsRequest request) {
-      LOGGER.debug("findPatientByUidEventHandler");
-      libMPI.startTransaction();
-      final var patient = libMPI.findPatientRecord(request.patientId);
-      request.replyTo.tell(new FindPatientRecordResponse(patient));
-      libMPI.closeTransaction();
+      PatientRecord patientRecord = null;
+      LOGGER.debug("findPatientRecordHandler");
+
+      try {
+         libMPI.startTransaction();
+         patientRecord = libMPI.findPatientRecord(request.patientId);
+         libMPI.closeTransaction();
+      } catch (Exception exception) {
+         LOGGER.error("libMPI.findPatientRecord failed for patientId: {} with error: {}",
+                      request.patientId,
+                      exception.getMessage());
+      }
+
+      if (patientRecord == null) {
+         request.replyTo.tell(new FindPatientRecordResponse(Either.left(new MpiServiceError.PatientIdDoesNotExistError(
+               "Patient not found",
+               request.patientId))));
+      } else {
+         request.replyTo.tell(new FindPatientRecordResponse(Either.right(patientRecord)));
+      }
+
       return Behaviors.same();
    }
 
    private Behavior<Event> findCandidatesHandler(final FindCandidatesRequest request) {
       LOGGER.debug("getCandidates");
       LOGGER.debug("{} {}", request.patientId, request.mu);
-      libMPI.startTransaction();
-      final var patient = libMPI.findPatientRecord(request.patientId);
-      final var recs = libMPI.getCandidates(patient.demographicData(), true);
-
-      CustomLinkerProbabilistic.updateMU(request.mu);
-      CustomLinkerProbabilistic.checkUpdatedMU();
-      final var candidates = recs
-            .stream()
-            .map(candidate -> new FindCandidatesResponse.Candidate(candidate,
-                                                                   CustomLinkerProbabilistic.probabilisticScore(candidate.demographicData(),
-                                                                                                                patient.demographicData())))
-            .toList();
-      request.replyTo.tell(new FindCandidatesResponse(Either.right(candidates)));
-      libMPI.closeTransaction();
+      PatientRecord patientRecord = null;
+      List<GoldenRecord> goldenRecords = null;
+      List<FindCandidatesResponse.Candidate> candidates = null;
+
+      try {
+         libMPI.startTransaction();
+         patientRecord = libMPI.findPatientRecord(request.patientId);
+         goldenRecords = libMPI.getCandidates(patientRecord.demographicData(), true);
+         libMPI.closeTransaction();
+      } catch (Exception exception) {
+         LOGGER.error("findCandidatesHandler failed to find patientId: {}", request.patientId);
+         request.replyTo.tell(new FindCandidatesResponse(Either.left(new MpiServiceError.PatientIdDoesNotExistError(
+               "Patient not found",
+               request.patientId))));
```

Comment:
I think this will be executed twice (see line 394)

Reply:
I think I forgot to remove this when refactoring. The only way to really know is to use a nested try catch. One for both libmpi calls then I can add the tell error to each catch
Are you happy with this approach?